Building with z88dk 2.3

I was looking to rebuild QE and ran into a couple of stumbling blocks with z88dk 2.3 (v22110-51889e5300-20231220) on MacOS.

The first is that -preserve doesn't seem to be a supported switch, and zcc tries to interpret it as a source file:

```
zcc +zxn -vn -O3 -startup=1 -clib=new -preserve -pragma-include:zpragma.inc -c    \
		--codesegBANK_47 --constsegBANK_47 \
		--datasegBANK_47 --bsssegBANK_47   \
		qe_banked.c -o./TMP/qe_banked.o
Unknown option2 -preserve
Cannot open source file: -preserve
make: *** [bank_47] Error 1
```
Removing that switch allows the build to proceed as far as the assemble stage, which fails to find QE_BANK_47.bin:

```
zxn: Error: Can't open "./TMP/QE_BANK_47.bin"
Page 95, allocate and loadBuilding application code failed
make: *** [assemble] Error 1
```

I've found that removing --clean from CZFLAGS allows the build to complete and the resulting QE binary works on the Next.

100% correct, this default behaviour changed between 2.1 and 2.2, please
feel free to submit a pull request, and I’ll get it merged in (although I
won’t update the build on the distro, there’s no harm in the code remaining
ever green <3 )

On Sun, 31 Dec 2023 at 13:32, Mike Quin ***@***.***> wrote:

> I was looking to rebuild QE and ran into a couple of stumbling blocks with
> z88dk 2.3 (v22110-51889e5300-20231220) on MacOS.
>
> The first is that -preserve doesn't seem to be a supported switch, and zcc
> tries to interpret it as a source file:
>
> zcc +zxn -vn -O3 -startup=1 -clib=new -preserve -pragma-include:zpragma.inc -c    \
> 		--codesegBANK_47 --constsegBANK_47 \
> 		--datasegBANK_47 --bsssegBANK_47   \
> 		qe_banked.c -o./TMP/qe_banked.o
> Unknown option2 -preserve
> Cannot open source file: -preserve
> make: *** [bank_47] Error 1
>
> Removing that switch allows the build to proceed as far as the assemble
> stage, which fails to find QE_BANK_47.bin:
>
> zxn: Error: Can't open "./TMP/QE_BANK_47.bin"
> Page 95, allocate and loadBuilding application code failed
> make: *** [assemble] Error 1
>
> I've found that removing --clean from CZFLAGS allows the build to complete
> and the resulting QE binary works on the Next.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/StalePixels/QEtoo/issues/1>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA4CH7AJLQTYNSEWVHHHVTYMFSNNAVCNFSM6AAAAABBIILW4GVHI2DSMVQWIX3LMV43ASLTON2WKOZSGA3DCMBWGEZTMNI>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

Pull request already here. You’re **too** good, Mike! x

On Sun, 31 Dec 2023 at 13:33, D. Rimron ***@***.***> wrote:

> 100% correct, this default behaviour changed between 2.1 and 2.2, please
> feel free to submit a pull request, and I’ll get it merged in (although I
> won’t update the build on the distro, there’s no harm in the code remaining
> ever green <3 )
>
> On Sun, 31 Dec 2023 at 13:32, Mike Quin ***@***.***> wrote:
>
>> I was looking to rebuild QE and ran into a couple of stumbling blocks
>> with z88dk 2.3 (v22110-51889e5300-20231220) on MacOS.
>>
>> The first is that -preserve doesn't seem to be a supported switch, and
>> zcc tries to interpret it as a source file:
>>
>> zcc +zxn -vn -O3 -startup=1 -clib=new -preserve -pragma-include:zpragma.inc -c    \
>> 		--codesegBANK_47 --constsegBANK_47 \
>> 		--datasegBANK_47 --bsssegBANK_47   \
>> 		qe_banked.c -o./TMP/qe_banked.o
>> Unknown option2 -preserve
>> Cannot open source file: -preserve
>> make: *** [bank_47] Error 1
>>
>> Removing that switch allows the build to proceed as far as the assemble
>> stage, which fails to find QE_BANK_47.bin:
>>
>> zxn: Error: Can't open "./TMP/QE_BANK_47.bin"
>> Page 95, allocate and loadBuilding application code failed
>> make: *** [assemble] Error 1
>>
>> I've found that removing --clean from CZFLAGS allows the build to
>> complete and the resulting QE binary works on the Next.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/StalePixels/QEtoo/issues/1>, or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAA4CH7AJLQTYNSEWVHHHVTYMFSNNAVCNFSM6AAAAABBIILW4GVHI2DSMVQWIX3LMV43ASLTON2WKOZSGA3DCMBWGEZTMNI>
>> .
>> You are receiving this because you are subscribed to this thread.Message
>> ID: ***@***.***>
>>
>

--- a/Makefile
+++ b/Makefile
@@ -14,8 +14,8 @@ Z88DK_CRT = 1
 DEBUGFLAGS =
 
 CC=zcc
-CCFLAGS=+zxn -vn -O3 -startup=$(Z88DK_CRT) -clib=new -preserve -pragma-include:zpragma.inc
-CZFLAGS=-Cz="--clean --fullsize --main-fence 0xC000"
+CCFLAGS=+zxn -vn -O3 -startup=$(Z88DK_CRT) -clib=new -pragma-include:zpragma.inc
+CZFLAGS=-Cz="--fullsize --main-fence 0xC000"
 LDFLAGS=-m -lm
 INCFLAGS=
 

